core/state_transition.go

```diff
@@ -260,7 +265,8 @@ func (st *StateTransition) buyGas() error {
 	st.gasRemaining += st.msg.GasLimit
 
 	st.initialGas = st.msg.GasLimit
-	st.state.SubBalance(st.msg.From, mgval)
+	mgvalU256, _ := uint256.FromBig(mgval)
```

Comment:
This can not overflow because mgval is guaranteed to be <= balanceCheck and we check that balance check does not overflow